src/zope/schema/_field.py

```diff
@@ -599,22 +613,24 @@ def _validate_fields(schema, value, errors=None):
     # that supports attribute assignment.)
     try:
         for name in schema.names(all=True):
-            if not IMethod.providedBy(schema[name]):
-                try:
-                    attribute = schema[name]
-                    if IChoice.providedBy(attribute):
-                        # Choice must be bound before validation otherwise
-                        # IContextSourceBinder is not iterable in validation
-                        bound = attribute.bind(value)
-                        bound.validate(getattr(value, name))
-                    elif IField.providedBy(attribute):
-                        # validate attributes that are fields
-                        attribute.validate(getattr(value, name))
-                except ValidationError as error:
-                    errors.append(error)
-                except AttributeError as error:
-                    # property for the given name is not implemented
-                    errors.append(SchemaNotFullyImplemented(error))
+            attribute = schema[name]
+            if IMethod.providedBy(attribute):
+                continue # pragma: no cover
+
+            try:
+                if IChoice.providedBy(attribute):
+                    # Choice must be bound before validation otherwise
+                    # IContextSourceBinder is not iterable in validation
+                    bound = attribute.bind(value)
+                    bound.validate(getattr(value, name))
+                elif IField.providedBy(attribute):
+                    # validate attributes that are fields
+                    attribute.validate(getattr(value, name))
+            except ValidationError as error:
+                errors[name] = error
+            except AttributeError as error:
+                # property for the given name is not implemented
+                errors[name] = SchemaNotFullyImplemented(error)
```

Comment:
I'm now wondering if this shouldn't have a `.with_field_and_value()` as well?

Reply:
I guess `attribute` is a field, so that makes sense. Value would always be `None` by definition.

Good point.